core/src/main/scala/com/databricks/labs/remorph/parsers/snowflake/rules/DealiasLCAFilter.scala

```diff
@@ -17,10 +17,9 @@ class DealiasLCAFilter extends Rule[LogicalPlan] with IRHelpers {
 
   def dealiasProject(project: Project, filter: Filter): Project = {
     val aliases = project.columns
-      .filter(col => col.isInstanceOf[Alias])
-      .map(col => col.asInstanceOf[Alias])
-      .filter(alias => alias.child.isInstanceOf[Id]) // TODO do we need to support more than that ?
-      .map(alias => alias.name.id -> alias.child.asInstanceOf[Id].id)
+      .collect { case a: Alias => a }
+      .filter(alias => !alias.child.isInstanceOf[Literal])
+      .map(alias => alias.name.id -> alias.child)
```

Comment:
```suggestion
      .collect { case Alias(e, name) if !e.isInstanceOf[Literal] => name.id -> e }
```

further distillation

Reply:
done